Getting Uncaught ReferenceError: THREE is not defined after following yeoman generator tutorial

I am getting these error messages on the browser console after following the tutorial.

```
ConsoleChannel.js:14 MapView: Failed to connect to datasource anonymous-datasource#1: Error during worker initialization: Uncaught ReferenceError: THREE is not defined in http://localhost:8080/decoder.bundle.js:35849

Uncaught ReferenceError: THREE is not defined
    at Object.three (external "THREE":1)
    at __webpack_require__ (bootstrap:18)
    at Object../node_modules/@here/harp-datasource-protocol/lib/ColorUtils.js (ColorUtils.js:10)
    at __webpack_require__ (bootstrap:18)
    at Object../node_modules/@here/harp-datasource-protocol/index.js (index.js:31)
    at __webpack_require__ (bootstrap:18)
    at Object../node_modules/@here/harp-vectortile-datasource/lib/VectorTileDecoder.js (VectorTileDecoder.js:9)
    at __webpack_require__ (bootstrap:18)
    at Object../node_modules/@here/harp-vectortile-datasource/index-worker.js (index-worker.js:18)
    at __webpack_require__ (bootstrap:18)


Uncaught ReferenceError: THREE is not defined
    at Object.three (external "THREE":1)
    at __webpack_require__ (bootstrap:18)
    at Object../node_modules/@here/harp-datasource-protocol/lib/ColorUtils.js (ColorUtils.js:10)
    at __webpack_require__ (bootstrap:18)
    at Object../node_modules/@here/harp-datasource-protocol/index.js (index.js:31)
    at __webpack_require__ (bootstrap:18)
    at Object../node_modules/@here/harp-vectortile-datasource/lib/VectorTileDecoder.js (VectorTileDecoder.js:9)
    at __webpack_require__ (bootstrap:18)
    at Object../node_modules/@here/harp-vectortile-datasource/index-worker.js (index-worker.js:18)
    at __webpack_require__ (bootstrap:18)
```

I am using v15.0.1 of nodejs, 6.14.8 of npm, 3.1.1 of yeoman and using WSL2(Ubuntu 18.04 LTS).

I have tried to update `npm `to` 7.0.8` since it can solve the problem of peer dependency, however I got this after running `npx -p yo -p @here/generator-harp.gl yo @here/harp.gl`

```
npm ERR! code ERESOLVE
npm ERR! Cannot read property 'length' of undefined

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/zzm/.npm/_logs/2020-11-05T23_27_28_496Z-debug.log
```

After running  `cat /home/zzm/.npm/_logs/2020-11-05T23_27_28_496Z-debug.log`

I got this

```
0 verbose cli [
0 verbose cli   '/home/zzm/.nvm/versions/node/v15.0.1/bin/node',
0 verbose cli   '/home/zzm/.nvm/versions/node/v15.0.1/bin/npm',
0 verbose cli   'install',
0 verbose cli   '--save-dev',
0 verbose cli   'webpack',
0 verbose cli   'webpack-cli',
0 verbose cli   'webpack-dev-server'
0 verbose cli ]
1 info using npm@7.0.8
2 info using node@v15.0.1
3 timing config:load:defaults Completed in 0ms
4 timing config:load:file:/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/npmrc Completed in 1ms
5 timing config:load:builtin Completed in 1ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 1ms
8 timing config:load:file:/home/zzm/test1/.npmrc Completed in 0ms
9 timing config:load:project Completed in 0ms
10 timing config:load:file:/home/zzm/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/home/zzm/.nvm/versions/node/v15.0.1/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 0ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 4ms
19 verbose npm-session 819bb6805dd25bd6
20 timing npm:load Completed in 9ms
21 timing arborist:ctor Completed in 0ms
22 timing idealTree:init Completed in 94ms
23 timing idealTree:userRequests Completed in 2ms
24 silly idealTree buildDeps
25 silly fetch manifest webpack@*
26 timing arborist:ctor Completed in 0ms
27 http fetch GET 304 https://registry.npmjs.org/webpack 186ms (from cache)
28 silly fetch manifest webpack-cli@*
29 http fetch GET 304 https://registry.npmjs.org/webpack-cli 32ms (from cache)
30 silly fetch manifest webpack-dev-server@*
31 http fetch GET 304 https://registry.npmjs.org/webpack-dev-server 30ms (from cache)
32 silly placeDep ROOT webpack-cli@4.2.0 OK for: test1@0.1.0 want: *
33 silly placeDep ROOT webpack-dev-server@3.11.0 OK for: test1@0.1.0 want: *
34 silly fetch manifest @webpack-cli/info@^1.1.0
35 silly fetch manifest @webpack-cli/serve@^1.1.0
36 silly fetch manifest colorette@^1.2.1
37 silly fetch manifest command-line-usage@^6.1.0
38 silly fetch manifest commander@^6.2.0
39 silly fetch manifest enquirer@^2.3.6
40 silly fetch manifest execa@^4.1.0
41 silly fetch manifest import-local@^3.0.2
42 silly fetch manifest interpret@^2.2.0
43 silly fetch manifest leven@^3.1.0
44 silly fetch manifest rechoir@^0.7.0
45 silly fetch manifest v8-compile-cache@^2.2.0
46 silly fetch manifest ansi-html@0.0.7
47 silly fetch manifest bonjour@^3.5.0
48 silly fetch manifest chokidar@^2.1.8
49 silly fetch manifest compression@^1.7.4
50 silly fetch manifest connect-history-api-fallback@^1.6.0
51 silly fetch manifest debug@^4.1.1
52 silly fetch manifest del@^4.1.1
53 silly fetch manifest express@^4.17.1
54 silly fetch manifest html-entities@^1.3.1
55 silly fetch manifest http-proxy-middleware@0.19.1
56 silly fetch manifest import-local@^2.0.0
57 silly fetch manifest internal-ip@^4.3.0
58 silly fetch manifest ip@^1.1.5
59 silly fetch manifest is-absolute-url@^3.0.3
60 silly fetch manifest killable@^1.0.1
61 silly fetch manifest loglevel@^1.6.8
62 silly fetch manifest opn@^5.5.0
63 silly fetch manifest p-retry@^3.0.1
64 silly fetch manifest portfinder@^1.0.26
65 silly fetch manifest schema-utils@^1.0.0
66 silly fetch manifest selfsigned@^1.10.7
67 silly fetch manifest serve-index@^1.9.1
68 silly fetch manifest sockjs@0.3.20
69 silly fetch manifest sockjs-client@1.4.0
70 silly fetch manifest spdy@^4.0.2
71 silly fetch manifest supports-color@^6.1.0
72 silly fetch manifest url@^0.11.0
73 silly fetch manifest webpack-dev-middleware@^3.7.2
74 silly fetch manifest webpack-log@^2.0.0
75 silly fetch manifest ws@^6.2.1
76 silly fetch manifest yargs@^13.3.2
77 http fetch GET 304 https://registry.npmjs.org/colorette 141ms (from cache)
78 http fetch GET 304 https://registry.npmjs.org/commander 146ms (from cache)
79 http fetch GET 304 https://registry.npmjs.org/execa 146ms (from cache)
80 http fetch GET 304 https://registry.npmjs.org/rechoir 148ms (from cache)
81 http fetch GET 304 https://registry.npmjs.org/command-line-usage 155ms (from cache)
82 http fetch GET 304 https://registry.npmjs.org/leven 155ms (from cache)
83 http fetch GET 304 https://registry.npmjs.org/import-local 156ms (from cache)
84 http fetch GET 304 https://registry.npmjs.org/v8-compile-cache 154ms (from cache)
85 http fetch GET 304 https://registry.npmjs.org/@webpack-cli%2finfo 163ms (from cache)
86 http fetch GET 304 https://registry.npmjs.org/interpret 159ms (from cache)
87 http fetch GET 304 https://registry.npmjs.org/bonjour 168ms (from cache)
88 http fetch GET 304 https://registry.npmjs.org/chokidar 171ms (from cache)
89 http fetch GET 304 https://registry.npmjs.org/compression 174ms (from cache)
90 http fetch GET 304 https://registry.npmjs.org/ansi-html 176ms (from cache)
91 http fetch GET 304 https://registry.npmjs.org/connect-history-api-fallback 185ms (from cache)
92 http fetch GET 304 https://registry.npmjs.org/debug 186ms (from cache)
93 http fetch GET 304 https://registry.npmjs.org/http-proxy-middleware 188ms (from cache)
94 http fetch GET 304 https://registry.npmjs.org/del 194ms (from cache)
95 http fetch GET 304 https://registry.npmjs.org/html-entities 195ms (from cache)
96 http fetch GET 304 https://registry.npmjs.org/express 196ms (from cache)
97 http fetch GET 304 https://registry.npmjs.org/import-local 204ms (from cache)
98 http fetch GET 304 https://registry.npmjs.org/internal-ip 203ms (from cache)
99 http fetch GET 304 https://registry.npmjs.org/is-absolute-url 208ms (from cache)
100 http fetch GET 304 https://registry.npmjs.org/ip 211ms (from cache)
101 http fetch GET 304 https://registry.npmjs.org/killable 214ms (from cache)
102 http fetch GET 304 https://registry.npmjs.org/schema-utils 219ms (from cache)
103 http fetch GET 304 https://registry.npmjs.org/p-retry 221ms (from cache)
104 http fetch GET 304 https://registry.npmjs.org/opn 223ms (from cache)
105 http fetch GET 304 https://registry.npmjs.org/portfinder 223ms (from cache)
106 http fetch GET 304 https://registry.npmjs.org/loglevel 226ms (from cache)
107 http fetch GET 304 https://registry.npmjs.org/@webpack-cli%2fserve 244ms (from cache)
108 http fetch GET 304 https://registry.npmjs.org/serve-index 232ms (from cache)
109 http fetch GET 304 https://registry.npmjs.org/selfsigned 234ms (from cache)
110 http fetch GET 304 https://registry.npmjs.org/spdy 234ms (from cache)
111 http fetch GET 304 https://registry.npmjs.org/sockjs 238ms (from cache)
112 http fetch GET 304 https://registry.npmjs.org/sockjs-client 239ms (from cache)
113 http fetch GET 304 https://registry.npmjs.org/webpack-dev-middleware 238ms (from cache)
114 http fetch GET 304 https://registry.npmjs.org/webpack-log 240ms (from cache)
115 http fetch GET 304 https://registry.npmjs.org/supports-color 246ms (from cache)
116 http fetch GET 304 https://registry.npmjs.org/ws 244ms (from cache)
117 http fetch GET 304 https://registry.npmjs.org/yargs 247ms (from cache)
118 http fetch GET 304 https://registry.npmjs.org/url 256ms (from cache)
119 http fetch GET 304 https://registry.npmjs.org/enquirer 394ms (from cache)
120 timing idealTree:#root Completed in 661ms
121 silly fetch manifest webpack-cli@4.x.x
122 http fetch GET 304 https://registry.npmjs.org/webpack-cli 28ms (from cache)
123 silly fetch manifest webpack@4.x.x || 5.x.x
124 http fetch GET 304 https://registry.npmjs.org/webpack 44ms (from cache)
125 silly placeDep ROOT @webpack-cli/info@1.1.0 OK for: webpack-cli@4.2.0 want: ^1.1.0
126 silly placeDep ROOT @webpack-cli/serve@1.1.0 OK for: webpack-cli@4.2.0 want: ^1.1.0
127 silly placeDep ROOT colorette@1.2.1 OK for: webpack-cli@4.2.0 want: ^1.2.1
128 silly placeDep ROOT command-line-usage@6.1.0 OK for: webpack-cli@4.2.0 want: ^6.1.0
129 silly placeDep node_modules/webpack-cli commander@6.2.0 OK for: webpack-cli@4.2.0 want: ^6.2.0
130 silly placeDep ROOT enquirer@2.3.6 OK for: webpack-cli@4.2.0 want: ^2.3.6
131 silly placeDep ROOT execa@4.1.0 OK for: webpack-cli@4.2.0 want: ^4.1.0
132 silly placeDep ROOT import-local@3.0.2 OK for: webpack-cli@4.2.0 want: ^3.0.2
133 silly placeDep ROOT interpret@2.2.0 OK for: webpack-cli@4.2.0 want: ^2.2.0
134 silly placeDep ROOT leven@3.1.0 OK for: webpack-cli@4.2.0 want: ^3.1.0
135 silly placeDep ROOT rechoir@0.7.0 OK for: webpack-cli@4.2.0 want: ^0.7.0
136 silly placeDep ROOT v8-compile-cache@2.2.0 OK for: webpack-cli@4.2.0 want: ^2.2.0
137 silly fetch manifest envinfo@^7.7.3
138 silly fetch manifest array-back@^4.0.0
139 silly fetch manifest chalk@^2.4.2
140 silly fetch manifest table-layout@^1.0.0
141 silly fetch manifest typical@^5.2.0
142 silly fetch manifest ansi-colors@^4.1.1
143 silly fetch manifest cross-spawn@^7.0.0
144 silly fetch manifest get-stream@^5.0.0
145 silly fetch manifest human-signals@^1.1.1
146 silly fetch manifest is-stream@^2.0.0
147 silly fetch manifest npm-run-path@^4.0.0
148 silly fetch manifest onetime@^5.1.0
149 silly fetch manifest signal-exit@^3.0.2
150 silly fetch manifest strip-final-newline@^2.0.0
151 silly fetch manifest resolve-cwd@^3.0.0
152 silly fetch manifest resolve@^1.9.0
153 http fetch GET 304 https://registry.npmjs.org/chalk 39ms (from cache)
154 http fetch GET 304 https://registry.npmjs.org/array-back 56ms (from cache)
155 http fetch GET 304 https://registry.npmjs.org/cross-spawn 55ms (from cache)
156 http fetch GET 304 https://registry.npmjs.org/ansi-colors 59ms (from cache)
157 http fetch GET 304 https://registry.npmjs.org/typical 61ms (from cache)
158 http fetch GET 304 https://registry.npmjs.org/table-layout 65ms (from cache)
159 http fetch GET 304 https://registry.npmjs.org/is-stream 66ms (from cache)
160 http fetch GET 304 https://registry.npmjs.org/npm-run-path 66ms (from cache)
161 http fetch GET 304 https://registry.npmjs.org/human-signals 68ms (from cache)
162 http fetch GET 304 https://registry.npmjs.org/onetime 67ms (from cache)
163 http fetch GET 304 https://registry.npmjs.org/get-stream 69ms (from cache)
164 http fetch GET 304 https://registry.npmjs.org/signal-exit 72ms (from cache)
165 http fetch GET 304 https://registry.npmjs.org/strip-final-newline 73ms (from cache)
166 http fetch GET 304 https://registry.npmjs.org/resolve 74ms (from cache)
167 http fetch GET 304 https://registry.npmjs.org/resolve-cwd 74ms (from cache)
168 http fetch GET 304 https://registry.npmjs.org/envinfo 312ms (from cache)
169 timing idealTree:node_modules/webpack-cli Completed in 405ms
170 silly placeDep ROOT envinfo@7.7.3 OK for: @webpack-cli/info@1.1.0 want: ^7.7.3
171 timing idealTree:node_modules/@webpack-cli/info Completed in 1ms
172 timing idealTree:node_modules/@webpack-cli/serve Completed in 0ms
173 timing idealTree:node_modules/colorette Completed in 0ms
174 silly placeDep ROOT array-back@4.0.1 OK for: command-line-usage@6.1.0 want: ^4.0.0
175 silly placeDep ROOT chalk@2.4.2 OK for: command-line-usage@6.1.0 want: ^2.4.2
176 silly placeDep ROOT table-layout@1.0.1 OK for: command-line-usage@6.1.0 want: ^1.0.0
177 silly placeDep ROOT typical@5.2.0 OK for: command-line-usage@6.1.0 want: ^5.2.0
178 silly fetch manifest ansi-styles@^3.2.1
179 silly fetch manifest escape-string-regexp@^1.0.5
180 silly fetch manifest supports-color@^5.3.0
181 silly fetch manifest deep-extend@~0.6.0
182 silly fetch manifest wordwrapjs@^4.0.0
183 http fetch GET 304 https://registry.npmjs.org/deep-extend 29ms (from cache)
184 http fetch GET 304 https://registry.npmjs.org/ansi-styles 38ms (from cache)
185 http fetch GET 304 https://registry.npmjs.org/supports-color 38ms (from cache)
186 http fetch GET 304 https://registry.npmjs.org/escape-string-regexp 46ms (from cache)
187 http fetch GET 304 https://registry.npmjs.org/wordwrapjs 46ms (from cache)
188 timing idealTree:node_modules/command-line-usage Completed in 51ms
189 timing idealTree:node_modules/array-back Completed in 0ms
190 silly placeDep ROOT ansi-styles@3.2.1 OK for: chalk@2.4.2 want: ^3.2.1
191 silly placeDep ROOT escape-string-regexp@1.0.5 OK for: chalk@2.4.2 want: ^1.0.5
192 silly placeDep node_modules/chalk supports-color@5.5.0 OK for: chalk@2.4.2 want: ^5.3.0
193 silly fetch manifest color-convert@^1.9.0
194 silly fetch manifest has-flag@^3.0.0
195 http fetch GET 304 https://registry.npmjs.org/has-flag 29ms (from cache)
196 http fetch GET 304 https://registry.npmjs.org/color-convert 31ms (from cache)
197 timing idealTree:node_modules/chalk Completed in 34ms
198 silly placeDep ROOT color-convert@1.9.3 OK for: ansi-styles@3.2.1 want: ^1.9.0
199 silly fetch manifest color-name@1.1.3
200 http fetch GET 304 https://registry.npmjs.org/color-name 32ms (from cache)
201 timing idealTree:node_modules/ansi-styles Completed in 33ms
202 silly placeDep ROOT color-name@1.1.3 OK for: color-convert@1.9.3 want: 1.1.3
203 timing idealTree:node_modules/color-convert Completed in 1ms
204 timing idealTree:node_modules/color-name Completed in 0ms
205 silly placeDep ROOT ansi-colors@4.1.1 OK for: enquirer@2.3.6 want: ^4.1.1
206 timing idealTree:node_modules/enquirer Completed in 1ms
207 timing idealTree:node_modules/ansi-colors Completed in 0ms
208 timing idealTree:node_modules/envinfo Completed in 0ms
209 timing idealTree:node_modules/escape-string-regexp Completed in 0ms
210 silly placeDep ROOT cross-spawn@7.0.3 OK for: execa@4.1.0 want: ^7.0.0
211 silly placeDep ROOT get-stream@5.2.0 OK for: execa@4.1.0 want: ^5.0.0
212 silly placeDep ROOT human-signals@1.1.1 OK for: execa@4.1.0 want: ^1.1.1
213 silly placeDep ROOT is-stream@2.0.0 OK for: execa@4.1.0 want: ^2.0.0
214 silly placeDep ROOT npm-run-path@4.0.1 OK for: execa@4.1.0 want: ^4.0.0
215 silly placeDep ROOT onetime@5.1.2 OK for: execa@4.1.0 want: ^5.1.0
216 silly placeDep ROOT signal-exit@3.0.3 OK for: execa@4.1.0 want: ^3.0.2
217 silly placeDep ROOT strip-final-newline@2.0.0 OK for: execa@4.1.0 want: ^2.0.0
218 silly fetch manifest path-key@^3.1.0
219 silly fetch manifest shebang-command@^2.0.0
220 silly fetch manifest which@^2.0.1
221 silly fetch manifest pump@^3.0.0
222 silly fetch manifest path-key@^3.0.0
223 silly fetch manifest mimic-fn@^2.1.0
224 http fetch GET 304 https://registry.npmjs.org/path-key 35ms (from cache)
225 http fetch GET 304 https://registry.npmjs.org/which 37ms (from cache)
226 http fetch GET 304 https://registry.npmjs.org/path-key 39ms (from cache)
227 http fetch GET 304 https://registry.npmjs.org/shebang-command 42ms (from cache)
228 http fetch GET 304 https://registry.npmjs.org/pump 42ms (from cache)
229 http fetch GET 304 https://registry.npmjs.org/mimic-fn 46ms (from cache)
230 timing idealTree:node_modules/execa Completed in 52ms
231 silly placeDep ROOT path-key@3.1.1 OK for: cross-spawn@7.0.3 want: ^3.1.0
232 silly placeDep ROOT shebang-command@2.0.0 OK for: cross-spawn@7.0.3 want: ^2.0.0
233 silly placeDep ROOT which@2.0.2 OK for: cross-spawn@7.0.3 want: ^2.0.1
234 silly fetch manifest shebang-regex@^3.0.0
235 silly fetch manifest isexe@^2.0.0
236 http fetch GET 304 https://registry.npmjs.org/shebang-regex 33ms (from cache)
237 http fetch GET 304 https://registry.npmjs.org/isexe 35ms (from cache)
238 timing idealTree:node_modules/cross-spawn Completed in 38ms
239 silly placeDep ROOT pump@3.0.0 OK for: get-stream@5.2.0 want: ^3.0.0
240 silly fetch manifest end-of-stream@^1.1.0
241 http fetch GET 304 https://registry.npmjs.org/end-of-stream 37ms (from cache)
242 timing idealTree:node_modules/get-stream Completed in 39ms
243 timing idealTree:node_modules/human-signals Completed in 0ms
244 silly placeDep ROOT resolve-cwd@3.0.0 OK for: import-local@3.0.2 want: ^3.0.0
245 silly fetch manifest resolve-from@^5.0.0
246 http fetch GET 304 https://registry.npmjs.org/resolve-from 38ms (from cache)
247 timing idealTree:node_modules/import-local Completed in 39ms
248 timing idealTree:node_modules/interpret Completed in 0ms
249 timing idealTree:node_modules/is-stream Completed in 0ms
250 timing idealTree:node_modules/leven Completed in 0ms
251 timing idealTree:node_modules/npm-run-path Completed in 0ms
252 silly placeDep ROOT mimic-fn@2.1.0 OK for: onetime@5.1.2 want: ^2.1.0
253 timing idealTree:node_modules/onetime Completed in 1ms
254 timing idealTree:node_modules/mimic-fn Completed in 0ms
255 timing idealTree:node_modules/path-key Completed in 0ms
256 silly placeDep ROOT end-of-stream@1.4.4 OK for: pump@3.0.0 want: ^1.1.0
257 timing idealTree:node_modules/pump Completed in 1ms
258 timing idealTree:node_modules/end-of-stream Completed in 0ms
259 silly placeDep ROOT resolve@1.18.1 OK for: rechoir@0.7.0 want: ^1.9.0
260 silly fetch manifest is-core-module@^2.0.0
261 silly fetch manifest path-parse@^1.0.6
262 http fetch GET 304 https://registry.npmjs.org/is-core-module 31ms (from cache)
263 http fetch GET 304 https://registry.npmjs.org/path-parse 32ms (from cache)
264 timing idealTree:node_modules/rechoir Completed in 33ms
265 silly placeDep ROOT is-core-module@2.1.0 OK for: resolve@1.18.1 want: ^2.0.0
266 silly placeDep ROOT path-parse@1.0.6 OK for: resolve@1.18.1 want: ^1.0.6
267 timing idealTree:node_modules/resolve Completed in 1ms
268 timing idealTree:node_modules/is-core-module Completed in 0ms
269 timing idealTree:node_modules/path-parse Completed in 0ms
270 silly placeDep ROOT resolve-from@5.0.0 OK for: resolve-cwd@3.0.0 want: ^5.0.0
271 timing idealTree:node_modules/resolve-cwd Completed in 1ms
272 timing idealTree:node_modules/resolve-from Completed in 0ms
273 silly placeDep ROOT shebang-regex@3.0.0 OK for: shebang-command@2.0.0 want: ^3.0.0
274 timing idealTree:node_modules/shebang-command Completed in 1ms
275 timing idealTree:node_modules/shebang-regex Completed in 0ms
276 timing idealTree:node_modules/signal-exit Completed in 0ms
277 timing idealTree:node_modules/strip-final-newline Completed in 0ms
278 silly placeDep ROOT deep-extend@0.6.0 OK for: table-layout@1.0.1 want: ~0.6.0
279 silly placeDep ROOT wordwrapjs@4.0.0 OK for: table-layout@1.0.1 want: ^4.0.0
280 silly fetch manifest reduce-flatten@^2.0.0
281 http fetch GET 304 https://registry.npmjs.org/reduce-flatten 28ms (from cache)
282 timing idealTree:node_modules/table-layout Completed in 31ms
283 timing idealTree:node_modules/deep-extend Completed in 0ms
284 timing idealTree:node_modules/typical Completed in 0ms
285 timing idealTree:node_modules/v8-compile-cache Completed in 0ms
286 silly fetch manifest webpack@^4.43.0
287 http fetch GET 304 https://registry.npmjs.org/webpack 36ms (from cache)
288 silly placeDep ROOT ansi-html@0.0.7 OK for: webpack-dev-server@3.11.0 want: 0.0.7
289 silly placeDep ROOT bonjour@3.5.0 OK for: webpack-dev-server@3.11.0 want: ^3.5.0
290 silly placeDep ROOT chokidar@2.1.8 OK for: webpack-dev-server@3.11.0 want: ^2.1.8
291 silly placeDep ROOT compression@1.7.4 OK for: webpack-dev-server@3.11.0 want: ^1.7.4
292 silly placeDep ROOT connect-history-api-fallback@1.6.0 OK for: webpack-dev-server@3.11.0 want: ^1.6.0
293 silly placeDep ROOT debug@4.2.0 OK for: webpack-dev-server@3.11.0 want: ^4.1.1
294 silly placeDep ROOT del@4.1.1 OK for: webpack-dev-server@3.11.0 want: ^4.1.1
295 silly placeDep ROOT express@4.17.1 OK for: webpack-dev-server@3.11.0 want: ^4.17.1
296 silly placeDep ROOT html-entities@1.3.1 OK for: webpack-dev-server@3.11.0 want: ^1.3.1
297 silly placeDep ROOT http-proxy-middleware@0.19.1 OK for: webpack-dev-server@3.11.0 want: 0.19.1
298 silly placeDep node_modules/webpack-dev-server import-local@2.0.0 OK for: webpack-dev-server@3.11.0 want: ^2.0.0
299 silly placeDep ROOT internal-ip@4.3.0 OK for: webpack-dev-server@3.11.0 want: ^4.3.0
300 silly placeDep ROOT ip@1.1.5 OK for: webpack-dev-server@3.11.0 want: ^1.1.5
301 silly placeDep ROOT is-absolute-url@3.0.3 OK for: webpack-dev-server@3.11.0 want: ^3.0.3
302 silly placeDep ROOT killable@1.0.1 OK for: webpack-dev-server@3.11.0 want: ^1.0.1
303 silly placeDep ROOT loglevel@1.7.0 OK for: webpack-dev-server@3.11.0 want: ^1.6.8
304 silly placeDep ROOT opn@5.5.0 OK for: webpack-dev-server@3.11.0 want: ^5.5.0
305 silly placeDep ROOT p-retry@3.0.1 OK for: webpack-dev-server@3.11.0 want: ^3.0.1
306 silly placeDep ROOT portfinder@1.0.28 OK for: webpack-dev-server@3.11.0 want: ^1.0.26
307 silly placeDep node_modules/webpack-dev-server schema-utils@1.0.0 OK for: webpack-dev-server@3.11.0 want: ^1.0.0
308 silly placeDep ROOT selfsigned@1.10.8 OK for: webpack-dev-server@3.11.0 want: ^1.10.7
309 silly placeDep ROOT serve-index@1.9.1 OK for: webpack-dev-server@3.11.0 want: ^1.9.1
310 silly placeDep ROOT sockjs@0.3.20 OK for: webpack-dev-server@3.11.0 want: 0.3.20
311 silly placeDep ROOT sockjs-client@1.4.0 OK for: webpack-dev-server@3.11.0 want: 1.4.0
312 silly placeDep ROOT spdy@4.0.2 OK for: webpack-dev-server@3.11.0 want: ^4.0.2
313 silly placeDep node_modules/webpack-dev-server supports-color@6.1.0 OK for: webpack-dev-server@3.11.0 want: ^6.1.0
314 silly placeDep ROOT url@0.11.0 OK for: webpack-dev-server@3.11.0 want: ^0.11.0
315 silly placeDep node_modules/webpack-dev-server webpack-dev-middleware@3.7.2 OK for: webpack-dev-server@3.11.0 want: ^3.7.2
316 timing idealTree Completed in 1595ms
317 timing command:install Completed in 1597ms
318 verbose stack Error: unable to resolve dependency tree
318 verbose stack     at Arborist.[failPeerConflict] (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1020:25)
318 verbose stack     at Arborist.[placeDep] (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1096:30)
318 verbose stack     at Arborist.[placeDep] (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1232:41)
318 verbose stack     at /home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:772:46
318 verbose stack     at Array.map (<anonymous>)
318 verbose stack     at Arborist.[buildDepStep] (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:772:8)
318 verbose stack     at async Arborist.buildIdealTree (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:203:7)
318 verbose stack     at async Promise.all (index 1)
318 verbose stack     at async Arborist.reify (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:121:5)
318 verbose stack     at async install (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/install.js:38:5)
319 verbose cwd /home/zzm/test1
320 verbose Linux 4.19.128-microsoft-standard
321 verbose argv "/home/zzm/.nvm/versions/node/v15.0.1/bin/node" "/home/zzm/.nvm/versions/node/v15.0.1/bin/npm" "install" "--save-dev" "webpack" "webpack-cli" "webpack-dev-server"
322 verbose node v15.0.1
323 verbose npm  v7.0.8
324 error code ERESOLVE
325 verbose stack TypeError: Cannot read property 'length' of undefined
325 verbose stack     at explainDependents (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/utils/explain-dep.js:68:19)
325 verbose stack     at explainNode (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/utils/explain-dep.js:13:3)
325 verbose stack     at explainEresolve (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/utils/explain-eresolve.js:24:24)
325 verbose stack     at report (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/utils/explain-eresolve.js:59:10)
325 verbose stack     at module.exports (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/utils/error-message.js:24:24)
325 verbose stack     at errorHandler (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/utils/error-handler.js:173:15)
325 verbose stack     at /home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/npm.js:118:9
325 verbose stack     at install (/home/zzm/.nvm/versions/node/v15.0.1/lib/node_modules/npm/lib/install.js:67:5)
326 verbose cwd /home/zzm/test1
327 verbose Linux 4.19.128-microsoft-standard
328 verbose argv "/home/zzm/.nvm/versions/node/v15.0.1/bin/node" "/home/zzm/.nvm/versions/node/v15.0.1/bin/npm" "install" "--save-dev" "webpack" "webpack-cli" "webpack-dev-server"
329 verbose node v15.0.1
330 verbose npm  v7.0.8
331 error Cannot read property 'length' of undefined
332 verbose exit 1
```

--- a/@here/generator-harp.gl/generators/app/templates/javascript/decoder.js
+++ b/@here/generator-harp.gl/generators/app/templates/javascript/decoder.js
@@ -4,8 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-self.importScripts("three.min.js");
-
+import "./deps";
 import {
     GeoJsonTilerService,
     VectorTileDecoderService

--- a/@here/generator-harp.gl/generators/app/templates/javascript/deps.js
+++ b/@here/generator-harp.gl/generators/app/templates/javascript/deps.js
@@ -1,0 +1,8 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+// This file is a simple work for webpack reordering the dependencies, i.e. imports are put
+// before importScripts.
+self.importScripts("three.min.js");

